### PR TITLE
fix(route): cast

### LINKED ISSUE #14587: 中国科协路由失效
### 路由地址

```routes
/cast/:column?
```

### 完整路由地址

```fullroutes
https://rsshub.app/cast
```

### 相关文档

https://docs.rsshub.app/routes/government#zhong-guo-ke-xue-ji-shu-xie-hui

### 预期是什么？

生成通知公告

### 实际发生了什么？

好像是中国科协更新了网址
新的网址为https://www.cast.org.cn/xw/tzgg/ZH/index.html

### 部署

RSSHub 演示 (https://rsshub.app)

### 部署相关信息

_No response_

### 额外信息

```shell
无
```

### 这不是重复的 issue

- [X] 我已经搜索了 [现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by KarasuShin**
## Involved Issue / 该 PR 相关 Issue

Close #14587

## Example for the Proposed Route(s) / 路由地址示例

```routes
/cast/qjkx/ZYWZ/2024
/cast/zk/zkdt
/cast/xs/XSDT
/cast/kp/KPDT
/cast/dj/DJYW
/cast/xw/SP
/cast/xw/SZYW
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [ ] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
修复中国科协路由失效

### CODE REVIEW

File: lib/routes/cast/index.ts [hunk 1]
```diff
@@ -1,71 +1,77 @@
...
-                title: a.text(),
-                pubDate: parseDate(item('.list-data').text().trim(), 'DDYYYY/MM'),
-                link: `${baseUrl}${a.attr('href')}`,
-            };
-        });
+    // 新闻-视频首页特殊处理
+    if (column === 'xw' && subColumn === 'SP' && !category) {
+        items = await parsePage(indexData);
+    } else {
+        const buildUnitScript = $('script').filter((_, el) => $(el).attr('parsetype') === 'bulidstatic');
```
Comment (TonyRL) [inline lib/routes/cast/index.ts:62]:
```suggestion
        const buildUnitScript = $('script[parseType="bulidstatic"]');
```

File: lib/routes/cast/index.ts [hunk 2]
```diff
@@ -1,71 +1,77 @@
...
+        const queryUrl = `${baseUrl}/${buildUnitScript.attr('url')}`;
```
Comment (TonyRL) [inline lib/routes/cast/index.ts:63]: Duplicated slash
https://github.com/DIYgod/RSSHub/actions/runs/8143752325/job/22256284076#step:12:27
```suggestion
        const queryUrl = `${baseUrl}${buildUnitScript.attr('url')}`;
```

File: website/docs/routes/government.mdx [hunk 3]
```diff
@@ -1348,7 +1348,21 @@
 
 ### 通用 {#zhong-guo-ke-xue-ji-shu-xie-hui-tong-yong}
 
-<Route author="TonyRL" example="/cast" path="/cast/:column?" paramsDesc={['栏目 ID，即 URL 中的数字，默认为 `457`']} radar="1" notOperational="1" />
+<Route author="KarasuShin TonyRL" example="/cast/xw/tzgg/ZH" path="/cast/:column/:subColumn/:category?" paramsDesc={['栏目编号，见下表', '二级栏目编号','分类']} radar="1" notOperational="1">
+  :::tip
+  在路由末尾处加上 `?limit=限制获取数目` 来限制获取条目数量，默认值为`10`
```
[the remark this answers is not in the document]
Comment (KarasuShin) [reply website/docs/routes/government.mdx:1353]: fixed in commit 7607c1550ee67f3388c6a093657f86216a1fce46

File: lib/routes/cast/router.ts [hunk 4]
```diff
@@ -1,3 +1,3 @@
 export default (router) => {
-    router.get('/:column?', './index');
+    router.get('/:column/:subColumn/:category?', './index');
```
[the remark this answers is not in the document]
Comment (KarasuShin) [reply lib/routes/cast/router.ts:2]: fixed in a6dfb5e5479817dc68e8d57d6534e2bbc41110bf

**Pull request merged**